helix-term/src/keymap.rs

```diff
@@ -62,252 +283,298 @@ impl DerefMut for Keymaps {
+            "A-`" => switch_to_uppercase,
+
+            "home" => goto_line_start,
+            "end" => goto_line_end,
+
+            "w" => move_next_word_start,
+            "b" => move_prev_word_start,
+            "e" => move_next_word_end,
+
+            "W" => move_next_long_word_start,
+            "B" => move_prev_long_word_start,
+            "E" => move_next_long_word_end,
+
+            "v" => select_mode,
+            "g" => { "Goto"
+                "g" => goto_file_start,
+                "e" => goto_file_end,
+                "h" => goto_line_start,
+                "l" => goto_line_end,
+                "s" => goto_first_nonwhitespace,
+                "d" => goto_definition,
+                "y" => goto_type_definition,
+                "r" => goto_reference,
+                "i" => goto_implementation,
+                "t" => goto_window_top,
+                "m" => goto_window_middle,
+                "b" => goto_window_bottom,
+                "a" => goto_last_accessed_file,
+            },
+            ":" => command_mode,
+
+            "i" => insert_mode,
+            "I" => prepend_to_line,
+            "a" => append_mode,
+            "A" => append_to_line,
+            "o" => open_below,
+            "O" => open_above,
             // [<space>  ]<space> equivalents too (add blank new line, no edit)
 
-
-            key!('d') => Command::delete_selection,
+            "d" => delete_selection,
             // TODO: also delete without yanking
-            key!('c') => Command::change_selection,
+            "c" => change_selection,
             // TODO: also change delete without yanking
 
-            // key!('r') => Command::replace_with_char,
-
-            key!('s') => Command::select_regex,
-            alt!('s') => Command::split_selection_on_newline,
-            key!('S') => Command::split_selection,
-            key!(';') => Command::collapse_selection,
-            alt!(';') => Command::flip_selections,
-            key!('%') => Command::select_all,
-            key!('x') => Command::extend_line,
-            key!('X') => Command::extend_to_line_bounds,
+            "s" => select_regex,
+            "A-s" => split_selection_on_newline,
+            "S" => split_selection,
+            ";" => collapse_selection,
+            "A-;" => flip_selections,
+            "%" => select_all,
+            "x" => extend_line,
+            "X" => extend_to_line_bounds,
             // crop_to_whole_line
 
+            "m" => { "Match"
+                "m" => match_brackets,
+                "s" => surround_add,
+                "r" => surround_replace,
+                "d" => surround_delete,
+                "a" => select_textobject_around,
+                "i" => select_textobject_inner,
+            },
+            "[" => { "Left bracket"
+                "d" => goto_prev_diag,
+                "D" => goto_first_diag,
+            },
+            "]" => { "Right bracket"
+                "d" => goto_next_diag,
```

Comment:
`"Left bracket"` and `"Right bracket"` sound awkward...

Reply:
Yeah we need a better name for these, they mimic unimpaired: https://github.com/tpope/vim-unimpaired

Reply:
I think the previous name for this is better.
```
┌select to next────────┐
│ d  diagnostic        │
│ D  diagnostic (last) │
└──────────────────────┘
```

Reply:
It won't always select though (and in this case it jumps, not selects), it's just a bunch of pair mappings. `[<space>` could be `o<esc>` for example